vNext/AISKU/snippet/snippet.js

```diff
@@ -1,96 +1,96 @@
-﻿var sdkInstance = "appInsightsSDK";
-window[sdkInstance] = "appInsights";
-var aiName = window[sdkInstance]; // provide non default instance name through key appInsightsSDK
-var aisdk = window[aiName] || (function (aiConfig) {
-    var appInsights = {
-        config: aiConfig
-    };
-    appInsights.initialize = true; // initialize sdk on download
+﻿(function() {
+    var aiName = "appInsights";
+    var aisdk = window[aiName] || (function (aiConfig) {
+        var appInsights = {
+            config: aiConfig
+        };
+        appInsights.initialize = true; // initialize sdk on download
 
-    // Assigning these to local variables allows them to be minified to save space:
-    var localDocument = document, localWindow = window, scriptText = "script";
-    setTimeout(function () {
-        var scriptElement = localDocument.createElement(scriptText);
-        scriptElement.src = aiConfig.url || "https://az416426.vo.msecnd.net/next/ai.2.min.js";
-        localDocument.getElementsByTagName(scriptText)[0].parentNode.appendChild(scriptElement);
-    });
+        // Assigning these to local variables allows them to be minified to save space:
+        var localDocument = document, localWindow = window, scriptText = "script";
+        setTimeout(function () {
+            var scriptElement = localDocument.createElement(scriptText);
+            scriptElement.src = aiConfig.url || "https://az416426.vo.msecnd.net/next/ai.2.min.js";
+            localDocument.getElementsByTagName(scriptText)[0].parentNode.appendChild(scriptElement);
```

Comment:
Can we change this as I don't believe it really doing what the intent around having the snippet is meant to do.

What I mean by this, is that this line will cause a new script tag to be dropped into the same place in the document as the first script (which should be this snippet AND should also be in the <head> section), this will cause this new script tag to become a blocking script under the following condition.
- If there is at least 1 more inline script after the execution of this snippet that is also in the <head> (i.e. a blocking script -- i.e. the critical path scripts)

We should really be adding this at the end of the document or at the very least (the easiest) within the <body> and not the <head>

Reply:
I think we want to block at least until the lazy methods are set up and the `onerror` handler is patched. If it is moved to the end document and becomes more "async", would we miss a lot of the initial telemetry from the pageload that our current blocking accomplishes? From my experience, users extremely sensitive to performance have either tweaked this snippet to their liking (using importScripts, etc) or made the containing `<script>` tag itself `async`.

Reply:
Spoke with Mark offline, this comment is about how we load the main script and not the snippet, so we wouldn't miss any telemetry etc as the snippet takes care of caching and replaying any reported telemetry between the snippet initialization and the loading of the main script.